cv-data-service-library/src/main/java/com/trihydro/library/helpers/TimGenerationHelper.java

```diff
@@ -1101,14 +1171,52 @@ private OdeGeoRegion getServiceRegion(List<Milepost> mileposts) {
         return serviceRegion;
     }
 
+    private Milepost attemptToCalculateAnchorPoint(TimUpdateModel tum,
+                                                   List<ResubmitTimException> exceptions,
+                                                   List<Milepost> allMps) {
+        Milepost firstPoint = allMps.get(0);
+        Milepost secondPoint = allMps.get(1);
+
+        try {
+            return getAnchorPoint(firstPoint, secondPoint);
+        } catch (Utility.IdenticalPointsException e) {
+            utility.logWithDate(
+                "Identical points found, attempting to use next two points for anchor");
+
+            if (allMps.size() < 3) {
+                String exMsg = String.format(
+                    "Unable to resubmit TIM, first two mileposts are identical for Active_Tim %d and less than 3 mileposts found",
+                    tum.getActiveTimId());
+                utility.logWithDate(exMsg);
+                exceptions.add(new ResubmitTimException(tum.getActiveTimId(), exMsg));
+                return null;
+            }
+
+            firstPoint = allMps.get(1);
+            secondPoint = allMps.get(2);
+
+            try {
+                return getAnchorPoint(firstPoint, secondPoint);
+            } catch (Utility.IdenticalPointsException e2) {
+                String exMsg = String.format(
+                    "First three mileposts are identical, unable to determine anchor for Active_Tim %d",
+                    tum.getActiveTimId());
+                utility.logWithDate(exMsg);
+                exceptions.add(new ResubmitTimException(tum.getActiveTimId(), exMsg));
+                return null;
+            }
```

Comment:
Clarification (non-blocking): If the first and second points are identical, the code will attempt to use the second and third points instead. If that also fails, it will return null.